trl/trainer/ppo_trainer.py

```diff
@@ -244,17 +244,41 @@ def __init__(
         else:
             self.kl_ctl = FixedKLController(self.config.init_kl_coef)
 
-        (
-            self.model,
-            self.ref_model,
-            self.optimizer,
-            self.data_collator,
-            self.dataloader,
-            self.lr_scheduler,
-        ) = self.accelerator.prepare(
-            self.model, self.ref_model, self.optimizer, self.data_collator, self.dataloader, self.lr_scheduler
+        # Safety checkers for DS integration
+        is_deepspeed_zero_3 = (
```

Comment:
This changes is irrespective of DS Stage, it should be applied for all DS Stages